### PR TITLE
SDIT-984 allocation migration int test

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/AllocationsMigrationMessageListener.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.concurrent.CompletableFuture
+
+@Service
+class AllocationsMigrationMessageListener(
+  objectMapper: ObjectMapper,
+  migrationService: AllocationsMigrationService,
+) : MigrationMessageListener<AllocationsMigrationFilter, FindActiveAllocationIdsResponse, GetAllocationResponse, AllocationMigrationMappingDto>(
+  objectMapper,
+  migrationService,
+) {
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/AllocationsMigrationMessageListener.kt:26]: The main code change is to wire up the message listener.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/AllocationsMigrationResource.kt [hunk 2]
```diff
@@ -24,7 +29,44 @@ import java.time.LocalDateTime
 @RequestMapping("/migrate", produces = [MediaType.APPLICATION_JSON_VALUE])
 class AllocationsMigrationResource(
   private val migrationHistoryService: MigrationHistoryService,
+  private val allocationsMigrationService: AllocationsMigrationService,
 ) {
+  @PreAuthorize("hasRole('ROLE_MIGRATE_ACTIVITIES')")
+  @PostMapping("/allocations")
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/AllocationsMigrationResource.kt:35]: And add the start migration endpoint

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/ActivitiesMigrationIntTest.kt [hunk 3]
```diff
@@ -81,98 +105,50 @@ class ActivitiesMigrationIntTest : SqsIntegrationTestBase() {
...
-      mappingApi.stubActivitiesMappingByMigrationId(count = 7)
-
-      webTestClient.post().uri("/migrate/activities")
-        .headers(setAuthorisation(roles = listOf("ROLE_MIGRATE_ACTIVITIES")))
-        .header("Content-Type", "application/json")
-        .body(BodyInserters.fromValue("""{ "prisonId": "BXI" }"""))
-        .exchange()
-        .expectStatus().isAccepted
+    fun `will migrate several pages of activities`() {
+      stubMigrationDependencies(7)
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/ActivitiesMigrationIntTest.kt:109]: The only changes to the existing actiivty migration tests is some refactoring and a tidy.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/AllocationMigrationIntTest.kt [hunk 4]
```diff
@@ -1,24 +1,243 @@
...
 import java.time.LocalDateTime
 
 class AllocationMigrationIntTest : SqsIntegrationTestBase() {
 
   @Autowired
   private lateinit var migrationHistoryRepository: MigrationHistoryRepository
 
+  @Nested
+  @DisplayName("POST /migrate/allocations")
+  inner class MigrateAllocations {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/AllocationMigrationIntTest.kt:43]: Of more interest are the allocation migration integration tests.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/ActivitiesMigrationIntTest.kt [hunk 5]
```diff
@@ -81,98 +105,50 @@ class ActivitiesMigrationIntTest : SqsIntegrationTestBase() {
...
-      nomisApi.stubGetInitialCount(ACTIVITIES_ID_URL, 1) { activitiesIdsPagedResponse(it) }
-      nomisApi.stubMultipleGetActivitiesIdCounts(totalElements = 1, pageSize = 3)
-      nomisApi.stubMultipleGetActivities(1..1)
-      mappingApi.stubAllMappingsNotFound(ACTIVITIES_GET_MAPPING_URL)
-      mappingApi.stubMappingCreate(ACTIVITIES_CREATE_MAPPING_URL)
-      activitiesApi.stubCreateActivityForMigration()
-      activitiesApi.stubGetActivityCategories()
-      mappingApi.stubActivitiesMappingByMigrationId(count = 1)
+    fun `will migrate a single course activity`() {
+      stubMigrationDependencies()
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/ActivitiesMigrationIntTest.kt:126]: Nice

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**